src/midi/midi.cpp

```diff
@@ -252,18 +254,24 @@ class MIDI : public Module_base {
 		LOG_MSG("MIDI: Can't find device: %s, using default handler.",
 		        dev.c_str());
 getdefault:
-		handler=handler_list;
-		while (handler) {
+		for (handler = handler_list; handler; handler = handler->next) {
+			const std::string name = handler->GetName();
+			if (name == "fluidsynth") {
+				// Never select fluidsynth automatically.
+				// Users needs to opt-in, otherwise
+				// fluidsynth will slow down emulator
+				// startup for all games.
+				continue;
+			}
 			if (handler->Open(conf)) {
 				midi.available=true;
 				midi.handler=handler;
-				LOG_MSG("MIDI: Opened device: %s",
-				        handler->GetName());
+				LOG_MSG("MIDI: Opened device: %s", name.c_str());
 				return;
 			}
-			handler=handler->next;
 		}
-		/* This shouldn't be possible */
+		assert(handler != nullptr);
+		assert(handler->GetName() == "none"s); // -V522
```

Comment:
The left hand side returns a char*, so we're not able to use std::string == operator from the "none"s string literal.

Swapping them, assert("none"s == handler->GetName()); should work but it less intuitive to read.

Maybe cast/wrap the LHS in a std::string?

Reply:
If `-V522` is a PVS suppression, we can add it to the `.pvs-suppress` file:
The PVS report contains a `general/suppression-list.json` file that has snippets for each detected issue. You can paste the snippet into `.pvs-suppress` and should take care of it (even if the code is moved around in future changes; it's pretty smart).

Reply:
Yeah, V522 was PVS warning - about the `handler` possibly being null (despite the assertion in line earlier). I combined these two assertions into one, and this resolved the warnings as well :)